aten/src/ATen/native/Gesv.cpp

```diff
@@ -0,0 +1,126 @@
+#include "ATen/ATen.h"
+#include "ATen/CPUApplyUtils.h"
+#include "ATen/Dispatch.h"
+#include "ATen/ExpandUtils.h"
+#include "ATen/NativeFunctions.h"
+
+#include "ATen/native/LinearAlgebraUtils.h"
+#include "ATen/native/Gesv.h"
+
+#include "TH.h"  // for USE_LAPACK
+
+#include <vector>
+
+#ifdef USE_LAPACK
+extern "C" void dgesv_(
+    int* n, int* nrhs, double* a, int* lda,
+    int *ipiv, double* b, int* ldb, int* info);
+extern "C" void sgesv_(
+    int* n, int* nrhs, float* a, int* lda,
+    int* ipiv, float* b, int* ldb, int* info);
+#endif
+
+namespace at { namespace native {
+
+template<class real>
+void lapackGesv(
+    int n, int nrhs, real* a, int lda, int* ipiv,
+    real* b, int ldb, int* info) {
+  AT_ERROR("gesv only takes float or double Tensors");
+}
+
+#ifdef USE_LAPACK
+template<> void lapackGesv<float>(
+    int n, int nrhs, float* a, int lda, int* ipiv,
+    float* b, int ldb, int* info) {
+  sgesv_(&n, &nrhs, a, &lda, ipiv, b, &ldb, info);
+}
+
+template<> void lapackGesv<double>(
+    int n, int nrhs, double* a, int lda, int* ipiv,
+    double* b, int ldb, int* info) {
+  dgesv_(&n, &nrhs, a, &lda, ipiv, b, &ldb, info);
+}
+#endif
+
+template <typename real>
+static void applyGesv(Tensor& b, Tensor& A, std::vector<int64_t> infos) {
+#ifndef USE_LAPACK
+  AT_ERROR("gesv: LAPACK library not found in compilation");
+#endif
+  real* A_data = (real*)A.data_ptr();
```

Comment:
would `.data<real>` work?

Reply:
Probably, most of the code from this PR is from before .data<real> was a thing. Let me try it (and I should also rename real to scalar_t)

Reply:
Yes, thanks!